include/cppoptlib/solver/bfgs.h

```diff
@@ -22,6 +24,12 @@ class Bfgs : public Solver<function_t> {
  public:
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
+  explicit Bfgs(const State<scalar_t> &stopping_state =
```

Comment:
Can't this ctor be part of the parent class?

Reply:
It is defined for `Solver`, but constructors aren't inherited. [Per cppreference](https://en.cppreference.com/w/cpp/language/using_declaration#Inheriting_constructors), I could expose the `Solver` constructor in the child classes by adding
```cpp
using Superclass::Superclass;
```
but I don't find this very readable or that saves much code. What do you think?